src/commands/open.ts

```diff
@@ -0,0 +1,101 @@
+import * as _ from "lodash";
+import * as clc from "cli-color";
+import * as open from "open";
+
+import * as FirebaseError from "../error";
+import * as api from "../api";
+import * as Command from "../command";
+import * as logger from "../logger";
+import { promptOnce } from "../prompt";
+import * as requirePermissions from "../requirePermissions";
+import * as requireInstance from "../requireInstance";
+import * as utils from "../utils";
+
+interface Link {
+  name: string;
+  arg: string;
+  consoleUrl?: string;
```

Comment:
I think consoleUrl is misleading - these are paths, can we call them paths? See below comment for restructuring these and refactoring special methods like `utils.consoleUrl` to replace with more general purpose ones that should make this whole logic simpler.

Reply:
Path is fine by me...